es-core/src/components/ImageGridComponent.h

```diff
@@ -1,15 +1,35 @@
 #pragma once
 
+#include "ThemeData.h"
 #include "GuiComponent.h"
 #include "components/IList.h"
 #include "components/ImageComponent.h"
+#include "components/TextComponent.h"
+#include "components/NinePatchComponent.h"
+#include "components/GridTileComponent.h"
+#include "Settings.h"
+#include "Renderer.h"
 #include "Log.h"
 
 struct ImageGridData
 {
 	std::shared_ptr<TextureResource> texture;
 };
 
+// Keeps track of which direction the user is moving.  ( for dynamic loading )
+enum UserDirection 
+{
+	MOVING_UP,
+	MOVING_DOWN
+};
+
+// A range around the cursor's index used for loading in textures
+struct CursorRange {
+	int min = 0;
+	int max = 12;
```

Comment:
I think it's set by default to 12 because we used a 3 * 4 grid at this time. It is dynamically adjusted at the end of the function updateLoadRange() anyway.